src/asl.ts

```diff
@@ -978,62 +1035,92 @@ export class ASL {
         })!;
       });
     } else if (isIfStmt(stmt)) {
-      return this.evalContextToSubState(stmt, (_, evalCondition) => {
-        const collect = (curr: IfStmt): [IfStmt[], Stmt | undefined] => {
-          if (curr._else) {
-            if (isIfStmt(curr._else)) {
-              const [ifs, el] = collect(curr._else);
-              return [[curr, ...ifs], el];
-            } else {
-              return [[curr], curr._else];
-            }
+      const collect = (curr: IfStmt): [IfStmt[], Stmt | undefined] => {
+        if (curr._else) {
+          if (isIfStmt(curr._else)) {
+            const [ifs, el] = collect(curr._else);
+            return [[curr, ...ifs], el];
+          } else {
+            return [[curr], curr._else];
           }
-          return [[curr], undefined];
-        };
-
-        const [ifs, els] = collect(stmt);
+        }
+        return [[curr], undefined];
+      };
 
-        const choices = ifs.map((_if, i) => ({
-          ...evalCondition(_if.when),
-          Next: `if_${i}`,
-        }));
+      const [ifs, els] = collect(stmt);
+
+      const ifStates: Record<string, ASLGraph.SubState> = Object.fromEntries(
+        ifs.map((_if, i) => {
+          const stateName = i === 0 ? "if" : `if_${i}`;
+          const next =
+            i + 1 < ifs.length
+              ? `if_${i + 1}`
+              : els
+              ? "else"
+              : ASLGraph.DeferNext;
+          const condition = this.eval(_if.when);
+          const conditionOutput = ASLGraph.getAslStateOutput(condition);
+          const stmtStates = this.evalStmt(_if.then, returnPass);
+          return [
+            stateName,
+            {
+              startState: "condition",
+              states: {
+                // run any states required for the condition and then evaluate the output (short circuit)
+                condition: ASLGraph.joinSubStates(_if.when, condition, {
```

Comment:
I can't grok `joinSubStates` - what do the first arguments mean? Are they other state graphs?

Reply:
The first argument is just a node to associate with the result to give the result a name.

I don't like it, but also don't like the idea of needing to do 

```ts
{
   ... ASLGraph.joinSubStates(...),
   node
}
```

everywhere...

or having to remove the varArgs from the function to move `node` to the end of the arguments.

So far now it is

1. a node to associate with the merged state.
2. one or more states to merge together into a sequence.

Reply:
Maybe rename the function `join` and take an object as input to help give meaning to the args?

Reply:
My plan was to rename this `sequence` or `chain` and get rid of `node` when figuring out better node naming.

Reply:
Sounds good - can be done later.

Reply:
#355 ^